web_poet/fields.py

```diff
@@ -63,6 +76,9 @@ def __init__(self, method):
                 raise TypeError(
                     f"@field decorator must be used on methods, {method!r} is decorated instead"
                 )
+            if out:
+                method = self._processed(method)
```

Comment:
Another nitpick: this looks a bit weird because of closures :) It's a bit surprising that self_processed can use `out` value. I think it can be solved either by passing `out` as an argument, or by moving this `if out:` check to the `self._processed` method.